### DISCUSSION TITLE
Milvus cluster with Dell ECS S3 replacing Minio

### DISCUSSION THREAD

**Discussion by Balaji5007** [milvus-io/milvus#31216] [Q&A and General discussion]
Would like to know if we can use the Dell ECS S3 instead of Minio for the cluster. If yes kindly share me the properties need to add in the docker compose file. I'm trying it for the milvus version 2.1.

**Comment by yhmo** [comment]
We didn't get a chance to test Dell ECS S3.
From its document seems most of S3 API are supported: https://www.dell.com/support/manuals/en-hk/ecs-appliance-software-with-encryption/ecs_pub_data_access_guide_3_3_to_3_6/s3-api-supported-and-unsupported-features?guid=guid-e21545d1-8529-4e72-a49a-ee0834a201a9&lang=en-us

Milvus is using minio-go sdk to interact with S3 storage services: https://min.io/docs/minio/linux/developers/go/API.html#
All the configuration about S3 is in the milvus.yaml: https://github.com/milvus-io/milvus/blob/1d962391378164e9024519590d6bb7bee60b9a90/configs/milvus.yaml#L66

You can try inputting Dell S3 endpoint in the "address", with the correct ak/sk. Leave the "cloudProvider" to be empty.
"bucketName" is the bucket name for storage. The ak/sk should have the authority to create a new bucket if the name doesn't exist.
"rootPath" is the root path under the bucket. All milvus data files will be stored under this path. Milvus will automatically create it if it doesn't exist.

By the way, version 2.1 is too old, and difficult to diagnose if you encounter bugs. We highly recommend using the latest v2.3.
